src/Common/OvercommitTracker.h

```diff
@@ -154,3 +154,14 @@ struct GlobalOvercommitTracker : OvercommitTracker
     DB::ProcessList * process_list;
     Poco::Logger * logger = &Poco::Logger::get("GlobalOvercommitTracker");
 };
+
+struct OvercommitTrackerBlockerInThread
```

Comment:
should it be non-copyable to prevent occasional misuse?

Reply:
It doesn't stores any data, so I don't see a problem here.

Reply:
> It doesn't stores any data, so I don't see a problem here.

Copied object may call dtor and decrement `counter` second time, lol